text/build_process.md

```diff
@@ -0,0 +1,146 @@
+
+Likewise, if somebody inadvertently breaks the API without increasing the major version (v0.5, in the example), we will break the other dependency. Thenceforth, we will need to run the tests of every actor using that dependency.
+
+#### Automate revisions bumps
+
+We could write a bumper script that will create a PR for each repo that depends on a bottom-level dependency and automatically lands it once all the PRs are green and an r+ was given on the initial PR. This seems to be the hardest solution in terms of initial set up. We will also rely more on Github than we currently do.
+
+### Have one repo with one single crate
+
+Bringing back taxonomy, thinkerbell and the adapters in foxbox will give a convenient way to find integration errors, as all the tests can be run locally without any particular inter-repository configuration.
+
+One major drawback would be the compile time. For more details about crate compile time, see [fxbox/foxbox#113](https://github.com/fxbox/foxbox/issues/113).
+
+### Have one main repo with many crates
+
+Everything that is tightly coupled or subject to heavy changes will live in the main repo. Other dependencies will remain in other repos. As of today, the repos which are candidates to be in the main repo are:
+* foxbox (aka the main repo)
+* openzwave-adapter
+* thinkerbell
+* taxonomy
+
+`cargo build` in the root directory will spot compile errors. A script to run `cargo test` in each submodule will be needed. If one would like to build/test only one module, we just need to cd to the right folder. Nonetheless, our CI jobs will likely take longer as more tests will need to be run in the same Travis job.
+
+Even if they are in the same repo, we can publish module on crates.io. E.g: [regex](https://github.com/rust-lang-nursery/regex) lives in the same repo as [regex_macros](https://github.com/rust-lang-nursery/regex/tree/master/regex_macros) and both are on crates.io (see [here](https://crates.io/crates/regex) and [there](https://crates.io/crates/regex_macros)).
+
+Foxbox would be the main entry point for contributors and we could describe the whole flow there.
+
+
+
+## Criterium/solution matrix
+
+| Criterium                  | Git submodules     | Point to master    | SemVer             | Automate bumps     | 1 repo / 1 crate   | 1 repo / many crates |
+| -------------------------- | ------------------ | ------------------ | ------------------ | ------------------ | ------------------ | -------------------- |
+| Fast compile time          | :white_check_mark: | :white_check_mark: | :white_check_mark: | :white_check_mark: | :x:                | :white_check_mark:   |
+| Easy dependency management | :x:                | :white_check_mark: | :white_check_mark: | :x:                | :white_check_mark: | :white_check_mark:   |
+| Easy rust up               | :x:                | :x:                | :x:                | :grey_question:    | :white_check_mark: | :white_check_mark:   |
```

Comment:
I think rusting up with git submodules is still substantially easier than doing it in several modules. It's a lot easier to automate by having everything in one repository (even with submodules).